android-sdk/src/main/java/ai/evolv/android_sdk/EvolvContextImpl.java

```diff
@@ -168,4 +189,242 @@ private void ensureInitialized() {
+        }
+
+        newArray.add(element);
+
+        for (int i = 0; i < newArray.size() ; i++) {
+            if(i > limit){
+                newArray.remove(i);
+            }
+        }
+        return set(key,newArray,local);
+    }
+
+    @Override
+    public boolean pushToArray(String key, String value, boolean local, int limit) {
+        ensureInitialized();
+        JsonObject context = local ? localContext : remoteContext;
+
+        JsonArray newArray = new JsonArray();
+        JsonObject element = new JsonObject();
+
+        JsonElement originalArray = helper.getValueForKey(key, context);
+
+        if(originalArray == null){
+            LOGGER.error("The " + "\"" + key + "\"" +  " does not have a suitable value ");
+            return false;
+        }else if(originalArray.isJsonPrimitive()){
+            element.addProperty(key,value);
+        }else if(originalArray.isJsonObject()){
+            element.add(key,originalArray);
+        }
+
+        newArray.add(element);
+
+        for (int i = 0; i < newArray.size() ; i++) {
+            if(i > limit){
+               newArray.remove(i);
+            }
+        }
+        return set(key,newArray,local);
+    }
+
+    @Override
+    public boolean contains(String key) {
+        ensureInitialized();
+        return remoteContext.has(key) || localContext.has(key);
+    }
+
+    // TODO: 26.07.2021 need to test!
+    @Override
+    public void update(JsonObject update, boolean local) {
+        ensureInitialized();
+        JsonObject context = local ? localContext : remoteContext;
+        flatten(update);
+
+        JsonObject flattenedBefore = new JsonObject();
+
+        for (String key : flattened.keySet()) {
+            flattenedBefore.add(key,context.get(key));
+        }
+
+        if (local) {
+
+            for (Map.Entry<String, JsonElement> entry : update.entrySet()) {
+                localContext.add(entry.getKey(),entry.getValue());
+            }
+            context = localContext;
+        } else {
+            for (Map.Entry<String, JsonElement> entry : update.entrySet()) {
+                remoteContext.add(entry.getKey(),entry.getValue());
+            }
+            context = remoteContext;
+        }
+
+        JsonObject updated = this.resolve();
+
+        for (String key : flattened.keySet()) {
+
+            if (flattenedBefore.get(key) == null) {
+
+                JsonObject object = new JsonObject();
+                object.addProperty("type",CONTEXT_INITIALIZED);
+                object.addProperty("key",key);
+                object.add("flattened_key",flattened.get(key));
+                object.addProperty("local",local);
+                object.add("updated",updated);
+
+                waitForIt.emit(this, CONTEXT_VALUE_ADDED, object);
+            }else if (flattenedBefore.get(key) != context.get(key)) {
+
+                JsonObject object = new JsonObject();
+                object.addProperty("type",CONTEXT_INITIALIZED);
```

Comment:
This type looks wrong -- would it not be CONTEXT_VALUE_CHANGED?

Reply:
absolutely right , fixed